File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:

        rightmax = len(height) - 1
        leftmax = 0 
        maxres = 0

        while leftmax < rightmax:
            hmax = min(height[leftmax] , height[rightmax])

            maxres = max(maxres , (rightmax - leftmax) * hmax)

            if height[leftmax] <= height[rightmax]:
                leftmax += 1
            else:
                rightmax -= 1
        return maxres
        

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length - 1;
        int resmax = 0;

        while(l < r)
        {
            int hmax = Math.min( height[l] , height[r]);
            resmax = Math.max(resmax , hmax * (r - l));

            if(height[l] > height[r])
            {
                r -= 1;
            }
            else
            {
                l += 1;
            }


        }
        return resmax;
        
    }
}